snuba/query/processors/timeseries_processor.py

```diff
@@ -185,3 +185,58 @@ def process_query(self, query: Query, request_settings: RequestSettings) -> None
         condition = query.get_condition_from_ast()
         if condition:
             query.set_ast_condition(condition.transform(self.__process_condition))
+
+
+GRANULARITY_MAPPING = {
+    "toStartOfMinute": 60,
+    "toStartOfHour": 3600,
+    "toDate": 86400,
+}
+
+
+def extract_granularity_from_query(query: Query, column: str) -> Optional[int]:
```

Comment:
Here you have a test for the timeseries processor. Which is the one that generates the bucketed expression starting from the granularity.
https://github.com/getsentry/snuba/blob/master/tests/query/processors/test_timeseries_processor.py#L138

I would advise you add, in the same test a call to this method and verify that the returned granularity is the right one. The test deals with conditions in the WHERE clause instead of the group by, but creating a new query to pass to your function that adds the transformed condition to the group by is trivial.